src/devices/Ft4222/FtFunction.cs

```diff
@@ -0,0 +1,561 @@
+        /// <returns>The status</returns>
+        [DllImport("LibFT4222.dll", CallingConvention = CallingConvention.Cdecl)]
+        public static extern FtStatus FT4222_I2CSlave_GetRxStatus(IntPtr ftHandle, ref ushort pRxSize);
+
+        /// <summary>
+        /// Operate an I2C read as a slave
+        /// </summary>
+        /// <param name="ftHandle">The handle of the open device</param>
+        /// <param name="buffer">The output read buffer</param>
+        /// <param name="bufferSize">The size of the buffer</param>
+        /// <param name="sizeTransferred">The size of the transfer</param>
+        /// <returns>The status</returns>
+        [DllImport("LibFT4222.dll", CallingConvention = CallingConvention.Cdecl)]
+        public static extern FtStatus FT4222_I2CSlave_Read(IntPtr ftHandle, byte[] buffer, ushort bufferSize, ref ushort sizeTransferred);
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="ftHandle">The handle of the open device</param>
+        /// <param name="buffer">The buffer to write</param>
+        /// <param name="bufferSize">The size of the buffer</param>
+        /// <param name="sizeTransferred">The size of the transfer</param>
+        /// <returns>The status</returns>
+        [DllImport("LibFT4222.dll", CallingConvention = CallingConvention.Cdecl)]
+        public static extern FtStatus FT4222_I2CSlave_Write(IntPtr ftHandle, byte[] buffer, ushort bufferSize, ref ushort sizeTransferred);
+
+        /// <summary>
+        /// Set I2C as a slave clock stretch
+        /// </summary>
+        /// <param name="ftHandle">The handle of the open device</param>
+        /// <param name="enable">True to enable, false to disable</param>
+        /// <returns>The status</returns>
+        [DllImport("LibFT4222.dll", CallingConvention = CallingConvention.Cdecl)]
+        public static extern FtStatus FT4222_I2CSlave_SetClockStretch(IntPtr ftHandle, bool enable);
+
+        /// <summary>
+        /// Set I2C as a slave response word
+        /// </summary>
+        /// <param name="ftHandle">The handle of the open device</param>
+        /// <param name="responseWord">The response word</param>
+        /// <returns>The status</returns>
+        [DllImport("LibFT4222.dll", CallingConvention = CallingConvention.Cdecl)]
+        public static extern FtStatus FT4222_I2CSlave_SetRespWord(IntPtr ftHandle, byte responseWord);
+
+        #endregion
+
+        #region GPIO
+
+        /// <summary>
+        /// Initialize the chi as GPIO
+        /// </summary>
+        /// <param name="ftHandle">The handle of the open device</param>
+        /// <param name="gpioDir">Array of pin configuration</param>
+        /// <returns>The status</returns>
+        [DllImport("LibFT4222.dll", CallingConvention = CallingConvention.Cdecl)]
+        public static extern FtStatus FT4222_GPIO_Init(IntPtr ftHandle, [MarshalAs(UnmanagedType.LPArray, SizeConst = 4)] GpioPinMode[] gpioDir);
```

Comment:
IIRC this should ByValArray

Reply:
I'm still trying to figure out what is the right way of marshaling this one. I've tried many things and it seems that it's not working. What puzzle me is that when calling this function, it returns that all is initialized but writing or reading doesn't change the status of the pins. So I'll change this one most likely